src/lib.rs

```diff
@@ -810,15 +821,24 @@ impl Builder {
     /// This makes bindgen generate enums instead of constants. Regular
     /// expressions are supported.
     ///
-    /// **Use this with caution.** You should not be using Rust enums unless
-    /// you have complete control of the C/C++ code that you're binding to.
-    /// Take a look at https://github.com/rust-lang/rust/issues/36927 for
-    /// more information.
+    /// **Use this with caution,** you probably want to use the non_exhaustive
+    /// flavor of rust enums instead of this one. Take a look at
+    /// https://github.com/rust-lang/rust/issues/36927 for more information.
     pub fn rustified_enum<T: AsRef<str>>(mut self, arg: T) -> Builder {
         self.options.rustified_enums.insert(arg);
         self
     }
 
+    /// Mark the given enum (or set of enums, if using a pattern) as a Rust
+    /// enum with the #[non_exhaustive] attribute.
+    ///
+    /// This makes bindgen generate enums instead of constants. Regular
+    /// expressions are supported.
+    pub fn rustified_enum_non_exhaustive<T: AsRef<str>>(mut self, arg: T) -> Builder {
```

Comment:
Oh, so this is what you meant with adding a new function to the `Builder`?

I was thinking more of a global switch like `fn generate_non_exhaustive(mut self, bool) -> Self {`, which just reuses the `rustified_enums` machinery based on the new config option... This is certainly more flexible, though a bit more complex. This is the reason for my comment in the issue:

> Something to potentially consider is whether it's likely people want some but not all enums to be non-exhaustive. But the initial version could just probably be an all-or-nothing switch.

I guess you implemented the hard version :)

Reply:
However, this approach means that we'll stop really respecting the `rust_target` configuration, which would be a bit unfortunate... We should probably at least integrate `non_exhaustive` with the rest of the rust target stuff (see `src/features.rs`), and assert / panic if `rustified_enums_non_exhaustive` is used with a `rust_target` that doesn't support it.

Reply:
If you consider the global switch approach a better solution, I could discard this PR and make a new one.